lib/Backends/CPU/CMakeLists.txt

```diff
@@ -9,43 +9,34 @@ find_program(LLVM_LINK_BIN
                llvm-link-6.0
                llvm-link)
 
-set(CMAKE_LLIR_CREATE_SHARED_LIBRARY "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
-set(CMAKE_LLIR_CREATE_SHARED_MODULE "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
-set(CMAKE_LLIR_CREATE_STATIC_LIBRARY "${LLVM_LINK_BIN} -o <TARGET> <OBJECTS>")
+set(CPURunttimeCompileOptions -ffast-math -g0 -emit-llvm -O0)
+set (CPURuntime_OBJS
+  ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit.bc.o
+  ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit_conv.bc.o
+  ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit_matmul.bc.o
+)
 
-add_library(CPURuntime
-              libjit/libjit.cpp
-              libjit/libjit_conv.cpp
-              libjit/libjit_matmul.cpp)
+# NOTE(abdulras) explicitly override the compiler and linker invocations with
+# custom rules.
+add_custom_command(
+  OUTPUT ${CPURuntime_OBJS}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime
+  COMMAND ${CLANG_BIN} -c libjit/libjit.cpp ${CPURunttimeCompileOptions} -o ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit.bc.o
+  COMMAND ${CLANG_BIN} -c libjit/libjit_conv.cpp ${CPURunttimeCompileOptions} -o ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit_conv.bc.o
+  COMMAND ${CLANG_BIN} -c libjit/libjit_matmul.cpp ${CPURunttimeCompileOptions} -o ${CMAKE_CURRENT_BINARY_DIR}/CPURuntime/libjit_matmul.bc.o
+  DEPENDS libjit/libjit.cpp libjit/libjit_conv.cpp libjit/libjit_matmul.cpp
+  WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
+)
 
-# NOTE(abdulras) explicitly override the compiler invocations with a custom
-# rule.  The trailing `#` is the comment leader intended to nullify the compile
-# commands.  Doing this allows us to override the compiler (driver) used for
-# building the runtime which requires clang (to emit LLVM IR for the LTO'ed AOT
-# JIT runtime).  Use a custom linker language with the rule specified above to
-# invoke our custom linker to merge the bitcode files.
-set_target_properties(CPURuntime
-                      PROPERTIES
-                        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
-                        CXX_STANDARD 11
-                        CXX_STANDARD_REQUIRED YES
-                        LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
-                        LINKER_LANGUAGE LLIR
-                        OUTPUT_NAME libjit.bc
-                        POSITION_INDEPENDENT_CODE YES
-                        PREFIX ""
-                        SUFFIX ""
-                        RULE_LAUNCH_COMPILE "${CLANG_BIN} <DEFINES> <INCLUDES> <FLAGS> -o <OBJECT> -c <SOURCE> #")
-target_compile_options(CPURuntime
-                       PRIVATE
-                         -ffast-math
-                         -g0
-                         -emit-llvm
-                         -O0)
+add_custom_target(CPURuntime
```

Comment:
Using `add_custom_target` means that CPURuntime will always get rebuilt, even if its deps have not changed.  Please try to find a way around that.

Reply:
I don't think it's quite correct. the CPURuntime will be evaluated, but it won't rebuild the .o files unless there is some change. Like they have been deleted., or the files that are compiles are modified. At least this is the behavior I am seeing on windows.

Reply:
@bertmaher Do you have a look at ayemolo's comment? There is no other way to building CPURuntime for Windows.